docs/naps/6-contributable-menus.md

```diff
@@ -0,0 +1,517 @@
+other types of cameras.
+
+### The I/O Utilities Menu
+
+A cursory analysis of widget names revealed a minimum of 17 plugins provide
+widgets dedicated to importing and exporting of data, features, models and/or
+other material supporting analysis.
+
+These widgets usually require more choices from the user than are currently
+possible via the `reader` and `writer` interfaces. Although many discussions have
+been raised about expanding the opening and saving options in `napari` to 
+support more complex choices ([#1637](https://github.com/napari/napari/issues/1637), 
+[#2801](https://github.com/napari/napari/issues/2801), 
+[#4611](https://github.com/napari/napari/issues/4611), 
+[#4882](https://github.com/napari/napari/pull/4882)...),
+we are not presently close to providing a unified interface for complex file opening/saving.
+Additionally, default `napari` opening and saving is entirely focused on reading data into 
+layer, but there are many other reasons a user may wish to read a file or save some output
+from the `Viewer`. 
+
+It is likely, therefore, that some plugins will always have bespoke interfaces for importing and
+exporting various file formats. These interfaces will be exposed via the new `File->I/O Utilities`
+menu.
+
+### Plugin Submenus
+
+The goal of the newly proposed menus is to provide a natural place where generally applicable
+actions can be semantically organized and easy to locate. However, many `napari` plugins
+contain an assortment of highly specialized widgets (that often interact with each other)
+that support highly specific, and sometimes ordered, workflows and analyses.
+
+It may never make sense for such plugins to distribute their widgets across the different
+`napari` menus, particularly when they are designed to work in concert on specific
+data formats or layer types.
+
+We therefore give plugin developers full control over their own submenu under `Plugins->My Plugin`.
+Plugin developers can organize all contributions under this submenu as they see fit, including adding their own submenus of arbitrary depth.
+
+### Complete Set of Proposed Contributable `napari` Menus
+
+```
+File
+├─ ...
+├─ IO Utilities
+Layers
+├─ Visualization
+├─ Edit
+│  ├─ Annotation Tools
+│  ├─ Filter
+│  ├─ Transform
+├─ Measure
+├─ Generate
+│  ├─ Registration
+│  ├─ Projection
+│  ├─ Segmentation
+│  ├─ Tracks
+│  ├─ Classification
+Acquisition
+```
+
+As a case study, we take four plugins offering between 9 and 14 widget contributions and arrange their widgets in these menus: 
+`empanada-napari`, `napari-stracking`, `napari-mm3` and `napari-clemreg`. Where a plugin's widgets don't
+naturally fit into one of the proposed menus, they are left in the plugin's own submenu.
+Note that we have arranged these widgets purely based on title and cursory inspection of the documentation, 
+so this should not be considered a concrete proposal for the structure of these plugins.
```

Comment:
It would be nice to also see how the `napari.yaml` would look like for these plugins. For example, it's not clear to me how exactly a plugin would opt for the plugin submenu versus the napari menus.

Also, should/can a plugin put a command in multiple places?

Reply:
> For example, it's not clear to me how exactly a plugin would opt for the plugin submenu versus the napari menus.

This mechanism does not yet exist, nor does the mechanism for a plugin declaring submenus for its widgets menu. I will work with @nclack to see what that might look like and provide an example before we accept the NAP.

> Also, should/can a plugin put a command in multiple places?

Plugins *can* do so currently. I do not personally believe they *should* - note I did not replicate widget contributions in the plugin's submenu in the example for that reason. I'm open to discussion on this, but I think given the command palette would always allow users to find everything a plugin offers, duplicating commands in menus would be confusing.